ansible/roles/shell/scripts/gm-mcp-ln.py

```
@@ -0,0 +1,87 @@
+    try:
+        with path.open("r", encoding="utf-8") as handle:
+            data = json.load(handle)
+    except FileNotFoundError as exc:
+        raise FileNotFoundError(f"Missing file: {path}") from exc
+    except (
+        json.JSONDecodeError
+    ) as exc:  # pragma: no cover - exercised via error path tests
+        raise ValueError(f"Invalid JSON in {path}: {exc}") from exc
+
+    if not isinstance(data, dict):
+        raise ValueError(f"Expected object at {path}")
```

Ruff (0.13.2)

Prefer `TypeError` exception for invalid type

(TRY004)